SimCalorimetry/HcalSimAlgos/src/HcalSiPMHitResponse.cc

```diff
@@ -307,3 +281,19 @@ double HcalSiPMHitResponse::generatePhotonTime(CLHEP::HepRandomEngine* engine) c
 double HcalSiPMHitResponse::Y11TimePDF(double t) {
   return exp(-0.0635-0.1518*t)*pow(t, 2.528)/2485.9;
 }
+
+void HcalSiPMHitResponse::setDetIds(const std::vector<DetId> & detIds) {
+  theDetIds = &detIds;
+
```

Comment:
Didn't fix it. :(


On 6/29/2016 3:17 PM, Kevin Pedro wrote:
>
> In SimCalorimetry/HcalSimAlgos/src/HcalSiPMHitResponse.cc 
> <https://github.com/cms-sw/cmssw/pull/15058#discussion_r69019089>:
>
> > @@ -307,3 +281,19 @@ double HcalSiPMHitResponse::generatePhotonTime(CLHEP::HepRandomEngine* engine) c
> >  double HcalSiPMHitResponse::Y11TimePDF(double t) {
> >    return exp(-0.0635-0.1518*t)*pow(t, 2.528)/2485.9;
> >  }
> > +
> > +void HcalSiPMHitResponse::setDetIds(const std::vector<DetId> & detIds) {
> > +  theDetIds = &detIds;
> > +
>
> You might need to put a check here in case of a zero-length vector...
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub 
> <https://github.com/cms-sw/cmssw/pull/15058/files/9571d588671cc059db43827e08c83e442b4c7631#r69019089>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe/AExICIPJKVtHCItW9KB9hzIk-sjNVLLgks5qQtLfgaJpZM4JBThX>.
>